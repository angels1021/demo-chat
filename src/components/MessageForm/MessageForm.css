.message-form textarea {
   resize: none;
}