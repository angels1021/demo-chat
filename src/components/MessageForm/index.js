/**
 * MessageForm component
 *
 * @module components/MessageForm
 * @component MessageForm
 * @typicalname MessageForm
 *
 * @requires [React]{@link https://facebook.github.io/react/docs/hello-world.html}
 * @requires [prop-types]{@link https://www.npmjs.com/package/prop-types}
 * */
import React from 'react';
import PropTypes from 'prop-types';
import { Form, InputGroup } from 'reactstrap';
import Submit from '../Submit';
import './MessageForm.css';

const MessageForm = ({ onSubmit }) => {
  let input;
  const SubmitForm = (e) => {
    e.preventDefault();
    if (!input.value) return;
    onSubmit(input.value);
    input.value = '';
  };
  return (
    <div className="message-form">
      <Form className="p-3" onSubmit={SubmitForm} >
        <InputGroup>
          <textarea className="form-control" ref={(node) => { input = node; }} />
          <div className="input-group-append" >
            <Submit>Send</Submit>
          </div>
        </InputGroup>
      </Form>
    </div>
  );
};

MessageForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default MessageForm;