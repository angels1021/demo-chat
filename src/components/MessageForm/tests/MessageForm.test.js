/**
 * Test MessageForm component
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MessageForm from '../';

const onSubmit = jest.fn();

const setup = () => shallow(<MessageForm onSubmit={onSubmit} />);
const wrapper = setup();

describe('<MessageForm>', () => {
  describe('default', () => {
    it('should render correctly with default options', () => {
      // assert
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('form actions', () => {
    const button = wrapper.find('Submit');

    it('should not fire on an empty input', () => {
      // act
      button.simulate('click');
      // assert
      expect(onSubmit).not.toHaveBeenCalled();
    });

    it('should fire the submit action', () => {
      //  arrange
      const input = wrapper.find('textarea');
      input.simulate('change', { target: { value: 'Hello' } })
      // act
      button.simulate('click');
      // assert
      expect(onSubmit).not.toHaveBeenCalledWith('Hello');
    });
  });

});
