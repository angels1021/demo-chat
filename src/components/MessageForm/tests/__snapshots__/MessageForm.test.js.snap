// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageForm> default should render correctly with default options 1`] = `
<div
  className="message-form"
>
  <Form
    className="p-3"
    onSubmit={[Function]}
    tag="form"
  >
    <InputGroup
      tag="div"
    >
      <textarea
        className="form-control"
      />
      <div
        className="input-group-append"
      >
        <Submit>
          Send
        </Submit>
      </div>
    </InputGroup>
  </Form>
</div>
`;
